frontend/src/components/chat/markdown-renderer.tsx

```diff
@@ -175,10 +166,11 @@ const COMPONENTS: Components = {
   code: ({ children, className }) => {
     const language = className?.replace("language-", "");
     if (language && typeof children === "string") {
+      const code = children.trim();
       return (
         <div>
           <div className="text-xs text-muted-foreground pl-1">{language}</div>
-          <CodeBlock code={children.trim()} language={language} />
```

Comment:
Oh, I see you are keying by the code, forcing an unmount. this might be more expensive since it needs to remount codemirror each time (which is not cheap)